timemachine/fe/free_energy.py

```diff
@@ -1168,10 +1168,13 @@ def compute_dense():
     return U_kl
 
 
-def verify_and_sanitize_potential_matrix(U_kl: NDArray, abs_energy_threshold: float = 1e9) -> NDArray:
+def verify_and_sanitize_potential_matrix(
+    U_kl: NDArray, hrex: HREX[CoordsVelBox], abs_energy_threshold: float = 1e9
+) -> NDArray:
     """Ensure energies in the diagonal are finite and below some threshold and sanitizes NaNs to infs."""
-    # Verify that the energies along the diagonal are finite, else a replica is no longer valid
-    replica_energies = np.diagonal(U_kl)
+    # Verify that the energies that the energies of the replica in the same state are finite, else a replica is no longer valid
+    state_idx = np.argsort(hrex.replica_idx_by_state)
+    replica_energies = np.diagonal(U_kl[:, state_idx])
```

Comment:
nit: is this equivalent? (I might be confused)

```suggestion
    replica_energies = np.diagonal(U_kl[hrex.replica_idx_by_state, :])
```

Reply:
nit: could this function just take `replica_idx_by_state` (I don't think it needs the whole `HREX` object?)